backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flare/internal/FlareShape.kt

```diff
@@ -0,0 +1,111 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.flare.internal
+
+import androidx.compose.ui.geometry.CornerRadius
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.geometry.RoundRect
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Outline
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.flare.BpkFlarePointerDirection
+import net.skyscanner.backpack.compose.flare.BpkFlareRadius
+import net.skyscanner.backpack.compose.tokens.BpkDimension
+
+internal class FlareShape(
+  private val radius: BpkFlareRadius,
+  private val pointerDirection: BpkFlarePointerDirection,
+) : Shape {
+
+  override fun createOutline(size: Size, layoutDirection: LayoutDirection, density: Density): Outline {
+    return Outline.Generic(
+      path = Path().apply {
+        val flareHeight = with(density) { FlareHeight.toPx() }
+        var scale = flareHeight / FlareVectorHeight
+        if (pointerDirection == BpkFlarePointerDirection.Up) {
+          scale *= -1
+        }
+        addRect(flareHeight, size, density, radius, pointerDirection)
+        addFlare(flareHeight, scale, size, pointerDirection)
+      }
+    )
+  }
+}
+
+private fun Path.addRect(
+  flareHeight: Float,
+  size: Size,
+  density: Density,
+  radius: BpkFlareRadius,
+  pointerDirection: BpkFlarePointerDirection,
+) {
+  val borderRect = when (pointerDirection) {
+    BpkFlarePointerDirection.Up -> Rect(0f, flareHeight, size.width, size.height + RectOffset)
```

Comment:
this was the easiest way to do this. add some more offset to the rect on the opposite side to avoid having to manually create the whole path